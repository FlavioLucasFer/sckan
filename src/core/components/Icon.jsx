import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

Icon.propTypes = {
	type: PropTypes.string,
	size: PropTypes.string,
	color: PropTypes.string,
	className: PropTypes.string,
	tooltipPosition: PropTypes.string,
	tooltip: PropTypes.string,
	onClick: PropTypes.func,
	children: PropTypes.string,
};

function Icon(props) {
	const { 
		type,
		size, 
		color, 
		className,
		tooltipPosition,
		tooltip,
		onClick,
		children,
		...rest
	} = props;

	useEffect(() => {
		M.Tooltip.init(document.querySelectorAll('.tooltipped'));
	}, []);

	function handleClick() {
		const elems = document.querySelectorAll('.tooltipped');

		for (let i = 0; i < elems.length; i++) {
			const tooltipInstance = M.Tooltip.getInstance(elems[i]);
			tooltipInstance.close();
		}

		if (onClick) {
			onClick();
		}
	}

	return (
		<i {...rest}
			className={`
				${tooltip && 'tooltipped'}
				${className || ''} ${size || ''} 
				${type ? `material-icons-${type}` : 'material-icons-outlined'}
				${color || ''}
			`}
			data-position={tooltipPosition || 'bottom'} 
			data-tooltip={tooltip || ''}
			onClick={handleClick}>
			{children}
		</i>
	);
}

export default Icon;