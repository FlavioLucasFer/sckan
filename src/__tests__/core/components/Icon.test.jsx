import React from 'react';
import { shallow } from 'enzyme';

import Icon from 'core/components/Icon';

describe('Testing Icon core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Icon>
				business
			</Icon>
		); 

		expect(wrapper.debug()).toMatchSnapshot();

		wrapper.setProps({
			type: 'round',
			size: 'large',
			color: 'grey-text',
			className: 'bold-text',
			tooptipPosition: 'left',
			tooptip: 'testing',
			onClick: () => console.log('icon test'),
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});