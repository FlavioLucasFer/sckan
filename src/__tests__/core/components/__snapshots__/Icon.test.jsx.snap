// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Icon core component should render correctly 1`] = `
"<i className=\\"\\\\n\\\\t\\\\t\\\\t\\\\tundefined\\\\n\\\\t\\\\t\\\\t\\\\t  \\\\n\\\\t\\\\t\\\\t\\\\tmaterial-icons-outlined\\\\n\\\\t\\\\t\\\\t\\\\t\\\\n\\\\t\\\\t\\\\t\\" data-position=\\"bottom\\" data-tooltip=\\"\\" onClick={[Function: handleClick]}>
  business
</i>"
`;

exports[`Testing Icon core component should render correctly 2`] = `
"<i tooptipPosition=\\"left\\" tooptip=\\"testing\\" className=\\"\\\\n\\\\t\\\\t\\\\t\\\\tundefined\\\\n\\\\t\\\\t\\\\t\\\\tbold-text large \\\\n\\\\t\\\\t\\\\t\\\\tmaterial-icons-round\\\\n\\\\t\\\\t\\\\t\\\\tgrey-text\\\\n\\\\t\\\\t\\\\t\\" data-position=\\"bottom\\" data-tooltip=\\"\\" onClick={[Function: handleClick]}>
  business
</i>"
`;
